### PR TITLE
Improvements to staticfile provider

### LINKED ISSUE #170: Static file root should be relative to app source
When specifying a directory to serve with the static file provider, you have to prefix with `/app`. The user should not have to know anything about the directory structure inside the docker container. Specifying `root: public` should actually serve `/app/public`

### PR DESCRIPTION
**Pull request by wyzlle**
- fixes #170 
 - Ability to configure root using environment variable
 - Auto serve from `public`, `index` or `dist` if found
 - fix `includes_file` API
 - add `includes_directory` API

### CODE REVIEW

**Review by JakeCooper — APPROVED** [review]
Awesome looks great!

File: src/providers/staticfile.rs [hunk 1]
```diff
@@ -116,3 +124,21 @@ impl Provider for StaticfileProvider {
         ))))
     }
 }
+
+impl StaticfileProvider {
+    pub fn get_root(app: &App, env: &Environment, staticfile_root: String) -> String {
+        let mut root = "";
+        if env.get_variable("NIXPACKS_STATICFILE_ROOT").is_some() {
+            root = env.get_variable("NIXPACKS_STATICFILE_ROOT").unwrap();
+        } else if staticfile_root.is_empty() {
```
Comment (coffee-cup) [inline src/providers/staticfile.rs:133]: Should this check if the root is not empty?

Comment (wyzlle) [reply src/providers/staticfile.rs:133]: Oops, My bad. Yeah, it should check if root is not empty.

**Review by devtomio — CHANGES_REQUESTED** [review]
shouldn't we use [`std::path::Path::is_file`](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.is_file) and [`std::path::Path::is_dir`](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.is_dir) as those don't return `Result`s

File: src/providers/staticfile.rs [hunk 2]
```diff
@@ -116,3 +124,21 @@ impl Provider for StaticfileProvider {
         ))))
     }
 }
+
+impl StaticfileProvider {
+    pub fn get_root(app: &App, env: &Environment, staticfile_root: String) -> String {
+        let mut root = "";
+        if env.get_variable("NIXPACKS_STATICFILE_ROOT").is_some() {
+            root = env.get_variable("NIXPACKS_STATICFILE_ROOT").unwrap();
```
Comment (devtomio) [inline src/providers/staticfile.rs:132]:
```suggestion
        if let Some(staticfile_root) = env.get_variable("NIXPACKS_STATICFILE_ROOT") {
            root = staticfile_root;
```

**Review by devtomio — CHANGES_REQUESTED** [review]

**Review by devtomio — APPROVED** [review]

**Review by coffee-cup — APPROVED** [review]

File: src/providers/staticfile.rs [hunk 3]
```diff
@@ -116,3 +124,21 @@ impl Provider for StaticfileProvider {
         ))))
     }
 }
+
+impl StaticfileProvider {
+    pub fn get_root(app: &App, env: &Environment, staticfile_root: String) -> String {
+        let mut root = "";
+        if let Some(staticfile_root) = env.get_variable("NIXPACKS_STATICFILE_ROOT") {
```
Comment (coffee-cup) [inline src/providers/staticfile.rs:131]: nit/for future: Can use `get_config_variable` that automatically prefixes with "NIXPACKS_"

Comment (wyzlle) [reply src/providers/staticfile.rs:131]: sure, lemme make that change.

Comment (Milo123459) [reply src/providers/staticfile.rs]: Changing how that function works should be in a different PR to update all the current uses of. NIXPACKS_ variables.

**Pull request merged**